runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/config/StreamingApplianceComputationConfigFetcherTest.java

```diff
@@ -0,0 +1,120 @@
+                    .setSystemName("systemName")
+                    .setStageName("stageName")
+                    .setInstructions(ImmutableList.of()));
+    Windmill.GetConfigResponse getConfigResponse =
+        Windmill.GetConfigResponse.newBuilder()
+            .addAllNameMap(nameMapEntries)
+            .addComputationConfigMap(
+                Windmill.GetConfigResponse.ComputationConfigMapEntry.newBuilder()
+                    .setComputationId("systemName")
+                    .setComputationConfig(
+                        Windmill.ComputationConfig.newBuilder()
+                            .addTransformUserNameToStateFamily(
+                                Windmill.ComputationConfig.TransformUserNameToStateFamilyEntry
+                                    .newBuilder()
+                                    .setStateFamily("stateFamilyName")
+                                    .setTransformUserName("transformUserName")
+                                    .build())
+                            .build())
+                    .build())
+            .addCloudWorks(serializedMapTask)
+            .build();
+    ComputationConfig expectedConfig =
+        ComputationConfig.create(
+            Transport.getJsonFactory().fromString(serializedMapTask, MapTask.class),
+            getConfigResponse.getComputationConfigMapList().stream()
+                .map(Windmill.GetConfigResponse.ComputationConfigMapEntry::getComputationConfig)
+                .flatMap(
+                    computationConfig ->
+                        computationConfig.getTransformUserNameToStateFamilyList().stream())
+                .collect(
+                    toMap(
+                        Windmill.ComputationConfig.TransformUserNameToStateFamilyEntry
+                            ::getTransformUserName,
+                        Windmill.ComputationConfig.TransformUserNameToStateFamilyEntry
+                            ::getStateFamily)),
+            nameMapEntries.stream()
+                .collect(
+                    toMap(
+                        Windmill.GetConfigResponse.NameMapEntry::getUserName,
+                        Windmill.GetConfigResponse.NameMapEntry::getSystemName)));
+    StreamingApplianceComputationConfigFetcher configLoader =
+        createStreamingApplianceConfigLoader();
+    when(mockWindmillServer.getConfig(any())).thenReturn(getConfigResponse);
+    Optional<ComputationConfig> config = configLoader.fetchConfig("someComputationId");
+    assertTrue(config.isPresent());
+    assertThat(config.get()).isEqualTo(expectedConfig);
+  }
+
+  @Test
+  public void testGetComputationConfig_whenNoConfigReturned() {
+    StreamingApplianceComputationConfigFetcher configLoader =
+        createStreamingApplianceConfigLoader();
+    when(mockWindmillServer.getConfig(any())).thenReturn(null);
+    Optional<ComputationConfig> configResponse = configLoader.fetchConfig("someComputationId");
+    assertFalse(configResponse.isPresent());
+  }
+
```

Comment:
add test that if the windmillfetcher throws exception that it is propagated to configLoader.fetchConfig